server/persistence/readclient/readclient.go

```diff
@@ -5,39 +5,18 @@ import (
 
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/cache"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/konflux-workspaces/workspaces/server/persistence/clientinterface"
 	icache "github.com/konflux-workspaces/workspaces/server/persistence/internal/cache"
 	"github.com/konflux-workspaces/workspaces/server/persistence/iwclient"
 	"github.com/konflux-workspaces/workspaces/server/persistence/mapper"
-
```

Comment:
I don't think we should move this interfaces out of this package. If this is needed in `mutate.ApplyHasDirectAccessLabel` we can define a new interface there. See my other comment

Reply:
I think the interfaces need to be extracted into a separate module that both `persistence/readclient` and `persistance/mutate` can import.  We can't define it in one or the other, since we end up with an import cycle that the go compiler _really_ dislikes.